libraries/ESP8266WebServer/src/ESP8266WebServer.h

```diff
@@ -80,6 +81,9 @@ class ESP8266WebServerTemplate
   using ClientType = typename ServerType::ClientType;
   using RequestHandlerType = RequestHandler<ServerType>;
   using WebServerType = ESP8266WebServerTemplate<ServerType>;
+  using ContentType_f = std::function<String(const String&)>;
+  enum ClientFuture_e { CLIENT_REQUEST_CAN_CONTINUE, CLIENT_REQUEST_IS_HANDLED, CLIENT_MUST_STOP, CLIENT_IS_GIVEN };
+  using Hook_f = std::function<ClientFuture_e(const String& method, const String& url, WiFiClient* client, ContentType_f contentType)>;
```

Comment:
I think we're still pulling all std::functions in to IRAM in the linker which might make this a problem.  Could you check on your own build where the webDAV hook ends up?

Reply:
Hooks are in FLASH (not IRAM). It is verified with your snippet and updated HelloServer example.